src/content-helper/dashboard-page/components/posts-table/component.tsx

```diff
@@ -0,0 +1,266 @@
+	page,
+} from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { Link } from 'react-router-dom';
+import { HydratedPost, QueryParams } from '../../../common/base-wordpress-provider';
+import { DashboardProvider } from '../../provider';
+
+/**
+ * PostInfo component.
+ *
+ * Represents the post information, the first column, in the PostsTable.
+ *
+ * @since 3.18.0
+ *
+ * @param {Object}       props      The component props.
+ * @param {HydratedPost} props.post The post object.
+ */
+const PostInfo = ( { post }: { post: HydratedPost } ): React.JSX.Element => {
+	const prettyDate = format( 'M j, o', post.date ?? '' );
+
+	return (
+		<div className="post-table--post-info">
+			<div className="thumbnail">
+				{ post.thumbnail ? (
+					<img src={ post.thumbnail } alt={ post.title.rendered } />
+				) : (
+					<div className="icon-container">
+						<Icon icon={ page } size={ 24 } />
+					</div>
+				) }
+			</div>
+			<div className="post-details">
+				<div className="post-title">
+					{ post.title.rendered !== ''
+						? post.title.rendered
+						: __( '(no title)', 'wp-parsely' ) }
+				</div>
+				<div className="post-meta">
+					<span className="post-date">{ prettyDate }</span>
+					<span className="post-author">{ post.author?.name }</span>
+					<div className="post-categories">
+						{ post.categories.map( ( category ) => (
+							<span key={ category.id }>{ category.name }</span>
+						) ) }
+					</div>
+				</div>
+			</div>
+		</div>
+	);
+};
+
+/**
+ * TablePagination component.
+ *
+ * Represents the pagination controls for the PostsTable.
+ *
+ * @since 3.18.0
+ *
+ * @param {Object}   props                The component props.
+ * @param {number}   props.currentPage    The current page.
+ * @param {Function} props.setCurrentPage The function to set the current page.
+ * @param {number}   props.totalPages     The total number of pages.
+ * @param {Function} props.onPrevious     The function to handle the previous button click.
+ * @param {Function} props.onNext         The function to handle the next button click.
+ */
+const TablePagination = ( {
+	currentPage,
+	setCurrentPage,
+	totalPages,
+	onPrevious,
+	onNext,
+}: {
+	currentPage: number;
+	setCurrentPage: ( value: number ) => void;
+	totalPages: number;
+	onPrevious: () => void;
+	onNext: () => void;
+} ): React.JSX.Element => {
+	return (
+		<div className="post-table--pagination-controls">
+			<div className="page-selector">
+				<span>{ __( 'Page', 'wp-parsely' ) }</span>
+				<NumberControl
+					value={ currentPage }
+					onChange={ ( value ) => {
+						let selectedPage = parseInt( value ?? '1', 10 );
+						if ( selectedPage > totalPages ) {
+							selectedPage = totalPages;
+						} else if ( selectedPage < 1 ) {
+							selectedPage = 1;
+						}
+
+						setCurrentPage( selectedPage );
+					} }
+					min={ 1 }
+					max={ totalPages }
+					dragDirection="e"
+				/>
+				<span>{ __( 'of', 'wp-parsely' ) } { totalPages }</span>
```

Comment:
Yeah, that's something I thought of. In all the languages that I know, I think it's possible to translate, but I'm not sure it will work for all the languages. Do you have any suggestion on how to tackle this?

Reply:
If it was all text, we could do something like a `sprintf` but here I see we've got the control in-between. If `DataViews` has the same paging text setup, we could look how it is done there, and if it's simple implement. Else I'm OK with this staying as is for the time being.